core/src/smartcontracts/isi/world.rs

```diff
@@ -303,7 +318,7 @@ pub mod isi {
             let parameter_id = parameter.id.clone();
 
             let world = &mut state_transaction.world;
-            if !world.parameters.remove(&parameter) {
+            if !world.parameters.swap_remove(&parameter) {
```

Comment:
in some cases ordering might matter. For parameters I think it's fine

Reply:
that being said, we should just use `BTreeSet` to avoid confusion

Reply:
`remove` is identical to `swap_remove`, just deprecated.